operator/controllers/helmrelease_controller_test.go

```diff
@@ -0,0 +1,157 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	shipitv1beta1 "ship-it-operator/api/v1beta1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	hapi "k8s.io/helm/pkg/proto/hapi/release"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type mockDownloader struct {
+	mock.Mock
+}
+
+func (m *mockDownloader) Download(ctx context.Context, chartName string) (*chart.Chart, error) {
+	args := m.Called(ctx, chartName)
+
+	var ret0 *chart.Chart
+	if args0 := args.Get(0); args0 != nil {
+		ret0 = args0.(*chart.Chart)
+	}
+
+	return ret0, args.Error(1)
+}
+
+var _ = Describe("HelmReleaseReconciler", func() {
+	ctx := context.Background()
+	log := ctrl.Log.WithName("helmrelease_controller_test")
+
+	releaseName := "test-release"
+	releaseNamespace := "test-namespace"
+
+	chart := &chart.Chart{
+		Metadata: &chart.Metadata{
+			Name: releaseName,
+		},
+	}
+
+	releaseKey := types.NamespacedName{
+		Name:      releaseName,
+		Namespace: releaseNamespace,
+	}
+
+	request := ctrl.Request{NamespacedName: releaseKey}
+
+	var (
+		helmClient  *helm.FakeClient
+		downloader  *mockDownloader
+		reconciler  *HelmReleaseReconciler
+		testRelease *shipitv1beta1.HelmRelease
+	)
+
+	BeforeEach(func() {
+		downloader = new(mockDownloader)
+		helmClient = new(helm.FakeClient)
+		reconciler = NewHelmReleaseReconciler(log, k8sClient, helmClient, downloader)
+
+		testRelease = &shipitv1beta1.HelmRelease{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "HelmRelease",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              releaseName,
+				Namespace:         releaseNamespace,
+				DeletionTimestamp: nil,
+				Annotations: map[string]string{
+					"helmreleases.shipit.wattpad.com/autodeploy": "true",
+				},
+			},
+			Spec: shipitv1beta1.HelmReleaseSpec{
+				ReleaseName: releaseName,
+				Chart: shipitv1beta1.ChartSpec{
+					Repository: "github.com/example/foo",
+					Path:       "bar/baz",
+				},
+				Values: runtime.RawExtension{Raw: []byte("{}")},
+			},
+			Status: shipitv1beta1.HelmReleaseStatus{
```

Comment:
Can just omit this whole thing.

Reply:
I'm not able to get the tests to run if `Conditions == nil`, with or without the `omitempty` struct tag. I get a k8s client error `validation failure list:\nstatus.conditions in body is required"`